File: WebDev101/learn_ruby/02_calculator/calculator.rb
#write your code here
def add(num1, num2)
  num1 + num2
end

def subtract(num1, num2)
  num1 - num2
end

def sum(arr)
  total = 0
  arr.each { |n| total += n }
  total
end

def multiply(*num)
  total = 1
  num.each { |n| total *= n}
  total
end

def power(num1, num2)
  num1 ** num2
end

def factorial(num)
  total = 1
  for n in 1..num
    total = total * n
  end
    total
end







# puts add(21, 8)

# puts subtract(21, 8)

# arr = 3.times.map do
#   rand(100)
# end

# puts arr
# puts "========="
# puts sum(arr)

# puts multiply(3, 3)

# puts power(2, 3)

# puts factorial(10)

File: TheOdinProject/01-WebDev101/05-JavascriptBasics/Fundamentals_Part_5/calculator/calculator.js
function add (a, b) {
	return a + b;
}

function subtract (a, b) {
	return a - b;
}

function sum (arr) {
  function getSum(total, num) {
    return total + num;
  };

	return arr.reduce(getSum, 0);
} 

function multiply (arr) {
	return arr.reduce((total, num) => total * num);
}

function power(b, e) {
	return Math.pow(b, e)
}

function factorial(n) {
  if (n == 0) return 1;
  
  let result = 1;

  for (let i = n; i > 0; i --) {
    result *= i;
  };
  return result
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}